src/diff_model.jl

```diff
@@ -47,15 +47,35 @@ function diff_model(_model::MOI.AbstractOptimizer)
         grads = []
         LHS = create_LHS_matrix(z, λ, Q, G, h, A)
         for param in params
```

Comment:
we should add a comment on what this loop on param is doing (what happens for each of the parameter types)

Reply:
added